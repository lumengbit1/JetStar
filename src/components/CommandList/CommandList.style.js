import styled from 'styled-components';
import { VFlex } from '../../styles/mixins';

export const Root = styled.div`
  ${VFlex};

  width: 32rem;
  height: 32rem;
  padding: 1rem;
  overflow: auto;
  background-color: black;
`;

export const Command = styled.div`
  color: yellow;
`;
