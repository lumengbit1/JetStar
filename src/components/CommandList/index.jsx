import React from 'react';
import { useSelector } from 'react-redux';
import { map } from 'lodash';
import { Root, Command } from './CommandList.style';

const CommandList = () => {
  const commands = useSelector((state) => state.reducer.commands);

  if (commands.length === 0) return null;

  return (
    <Root aria-label="command-list">
      {map(commands, (command, index) => (
        <Command
          key={`${command} + ${index}`}
        >
          {command}
        </Command>
      ))}
    </Root>
  );
};

export default CommandList;
