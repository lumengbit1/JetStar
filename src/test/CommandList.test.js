import React from 'react';
import 'jest-styled-components';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';

import { render } from '@testing-library/react';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store'
import CommandList from '../components/CommandList';

const mockStore = configureMockStore([thunk]);

describe('CommandList Test Cases', () => {
  it('1: Component shoule be rendered correctly', async () => {
    const store = mockStore({
      reducer: { commands: 'MOVE' }
    });

    const { container } = render(
      <Provider store={store}>
        <CommandList />
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });
});