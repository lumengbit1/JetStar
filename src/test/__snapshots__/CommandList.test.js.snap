// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommandList Test Cases 1: Component shoule be rendered correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: black;
  width: 32rem;
  height: 32rem;
  padding: 1rem;
  overflow: auto;
}

.c1 {
  color: yellow;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      M
    </div>
    <div
      class="c1"
    >
      O
    </div>
    <div
      class="c1"
    >
      V
    </div>
    <div
      class="c1"
    >
      E
    </div>
  </div>
</div>
`;
